File: 7KYU/Reverse Words/solution.cs
using System;
using System.Collections.Generic;
using System.Linq;
public static class Kata
{
	public static string ReverseWords(string str)
	{
		var split = str.Split(' ');
		var reversed = new List<string>();
		foreach (var arr in split.Select(s => s.ToCharArray()))
		{
			Array.Reverse(arr);
			var foo = new string(arr);
			reversed.Add(foo);
		}
		return string.Join(" ", reversed);
	}
}

File: 7KYU/Reverse Words/solution.go
package kata


import (
	"strings"
)

func Reverse(s string) string {
  xs := []rune(s)

  for i, j := 0, len(xs)-1; i < j; i, j = i+1, j-1 {
		xs[i], xs[j] = xs[j], xs[i]
	}

  return string(xs)
}

func ReverseWords(str string) string {
  words := strings.Split(str, " ")

  for idx, w := range words {
    words[idx] = Reverse(w)    
  }
    
  return strings.Join(words, " ")
}